src/components/layout.tsx

```diff
@@ -23,8 +23,11 @@ const Layout = ({ children, title, description, img }: Props) => {
     if ('IntersectionObserver' in window) {
       setupObserver();
     } else {
-      // Fallback for browsers without IntersectionObserver support
-      magicHeroNumber();
+      // Use polyfill for browsers without IntersectionObserver support
+      import('intersection-observer')
+        .then(setupObserver)
+        // Fallback for browsers without IntersectionObserver support
```

Comment:
```suggestion
        // Fallback to legacy CSS if polyfill fails to load
```

Reply:
Can I jump in here first (just joining the efforts)…

Are we saying that the only issue for the polyfill here is the missing types?